src/DotLiquid/Context.cs

```diff
@@ -585,23 +589,28 @@ private object LookupAndEvaluate(object obj, object key)
         {
             object value;
             if (obj is IDictionary dictionaryObj)
-            { 
+            {
                 value = dictionaryObj[key];
             }
+            // Resolve #350/#417, add support for rendering of a nested  ExpandoObject
+            else if (obj is IDictionary<string, object> dictionaryObject)

```

Comment:
Does this code replace the need for similar type checking and recursive Hash.FromDictionary calls?

Reply:
I can take a look for recursive Hash.FromDictionary calls, did you have any specific locations in mind using similar type checking?

Reply:
I am referring to this block:
```
            foreach (var keyValue in dictionary)
            {
                    if (keyValue.Value is IDictionary<string, object> innerDictionary)
                    {
                        result.Add(keyValue.Key, FromDictionary(innerDictionary));
                    }
                    else
                    {
                        result.Add(keyValue);
                    }
            }
```
It can be simplified after your change:
```
            foreach (var keyValue in dictionary)
            {
                    result.Add(keyValue);
            }
```

Reply:
@microalps, whilst looking at this I've found what I think is a pre-existing bug (#442) in for loops when a Hash is created with nested Dictionary objects.
I'll submit a separate PR for that bug fix.

Reply:
@microalps ok to resolve this conversation here then?

Reply:
Resolves #442